### ISSUE TITLE
attack with "textattack/distilbert-base-uncased-imdb" produces perturbations with the same predictions as the original data

### ISSUE THREAD

**Issue by ohadOrbach** [QData/TextAttack#277]
hi,
i made more than 10K attacks of **PWWS** and more than 20K attacks of **textfooler** on IMDB dataset.

when i used **BertForSequenceClassification** model with "pretrained" from "**textattack/distilbert-base-uncased-imdb**"
the attack did not worked.
for the base sentence and the the attacked sentence i got **the same prediction for 98% of the data.**

any ideas why?

**Comment by jxmorris12** [comment]
Hi @ohadOrbach -- sounds like something is definitely wrong! Let's get a little more information.

What command did you run? Did you use a model wrapper? Can you share with us the bash command or Python code you invoked to try PWWS and textfooler? Thanks!

**Comment by ohadOrbach** [comment]
Hi @jxmorris12 

1. i didn't use a model wrapper

2. the python code i used for generating attacks is
!textattack attack --model distilbert-base-uncased-imdb --recipe pwws --num-examples 10000 --shuffle False --checkpoint-interval 100 --csv-style plain  --log-to-csv AttackDatsetPWWSRen2019

3. the data i used for the model looks like the following:

text1 | text2 | label
-- | -- | --
The plot: Michael Linnett Connors has done eve... **|** The plot: Michael Linnett Connors has done eve... **|** 1.0
It's 1913. A studio prop boy spies the actress... **|** It's 1913. A studio prop boy spies the actress... **|** 1.0
What makes this one better than most "movie mo... **|** What makes this one better than most "movie mo... **|** 1.0
I can remember seeing this movie when I was ve... **|** I can remember seeing this movie when I was ve... **|** 1.0
Saw this film the first time in 1953 with my o... **|** byword this film the low time in 1953 with my ... **|** 1.0

4. the validation code is:

```
def validate_pairs():
    model.eval()
    fin_text1=[]
    fin_text2=[]
    with torch.no_grad():
        for data in tqdm(all_pairs_loader, 0):
            text1_ids = data["text1_ids"].to(device, dtype=torch.long)
            text1_mask = data["text1_mask"].to(device, dtype=torch.long)
            text1_token_type_ids = data["text1_token_type_ids"].to(device, dtype=torch.long)

            text1_outputs = model(text1_ids, text1_mask, text1_token_type_ids)         

            fin_text1.extend(torch.sigmoid(text1_outputs[0]).cpu().detach().numpy().tolist())
            
            text2_ids = data["text2_ids"].to(device, dtype=torch.long)
            text2_mask = data["text2_mask"].to(device, dtype=torch.long)
            text2_token_type_ids = data["text2_token_type_ids"].to(device, dtype=torch.long)

            text2_outputs = model(text2_ids, text2_mask, text2_token_type_ids)
     
            fin_text2.extend(torch.sigmoid(text2_outputs[0]).cpu().detach().numpy().tolist())

    return fin_text1, fin_text2
```

```
def agreed_labels(labels1, labels2):
  labels1 = np.array(labels1)
  labels2 = np.array(labels2)

  labels1[labels1 < 0.5] = 0
  labels1[labels1 >= 0.5] = 1
  labels1=labels1[:,0]
  
  labels2[labels2 < 0.5] = 0
  labels2[labels2 >= 0.5] = 1
  labels2=labels2[:,0]
  
  agree = (labels1 == labels2)
  
  return agree    
```


5. the "agreed" labels was 98% of all the data

**Comment by jxmorris12** [comment]
Hi @ohadOrbach, I tried running this in a Colab notebook; [here's a link](https://colab.research.google.com/drive/1WxdCoHchLxA0lz5ZkrP33w0mtCxJ3ZOy?usp=sharing).

I believe the issue is `--shuffle=False`. The IMDB dataset lists examples grouped by class: first, all the negative samples, then, all the positive samples. If you don't shuffle, you'll have to attack all the positive samples before getting to the negative ones.

Since you specified `--shuffle=False`, pretty much all of the model's predictions are positive, which is the intended behavior.

Let me know if you have questions!

**Comment by ohadOrbach** [comment]
Hi @jxmorris12 , [here ](https://colab.research.google.com/drive/1OQGoCtp9PIXfom2CuiUX1OutnA7ctEIg?usp=sharing)is the Colab notebook

I don't think this is the problem here.

I used the model only to **predict** with this data and **not to be trained** by this data.

The model is pretrained with "textattack/distilbert-base-uncased-imdb" and only the evaluation made with this data.

**Comment by jxmorris12** [comment]
please provide access! @ohadOrbach

**Comment by ohadOrbach** [comment]
Sorry try this one
 [here](https://colab.research.google.com/drive/179CWM0pA8bE-bmfGuoJeoq3NUerQ0GK1?usp=sharing)

**Comment by jxmorris12** [comment]
@ohadOrbach there's something wrong with your model and/or tokenizer loading code. All the predictions in the notebook are right around 0.5. This is more likely to be tokenizer misconfiguration than a model issue. Since the attack ran correctly, the model works correctly in TextAttack.

**Comment by jxmorris12** [comment]
You should also be using softmax instead of sigmoid, but that's not the issue

**Comment by ohadOrbach** [comment]
Yes you are defiantly right.
Thank you for the quick response and help!

**Comment by jxmorris12** [comment]
No problem @ohadOrbach! Happy to help.
